### PR TITLE
Text input method

### PR DESCRIPTION
**Pull request by chatrapathi06**

### CODE REVIEW

File: ReactSkia/components/RSkComponent.h [hunk 1]
```diff
@@ -87,7 +87,7 @@ class RSkComponent : public RnsShell::Layer , public SpatialNavigator::Container
      /* TODO Return default None here when state update is handled with proper mask */
      return RnsShell::LayerInvalidateAll;
   };
-  virtual void handleCommand(std::string commandName,folly::dynamic args){};
+  virtual void handleCommand(std::string commandName,folly::dynamic args){RNS_LOG_INFO("In Component");};
```
Comment (pprabhu-ng) [inline ReactSkia/components/RSkComponent.h]: is this change required ?
If yes , then need to add proper logging message

File: ReactSkia/components/RSkComponentTextInput.cpp [hunk 2]
```diff
@@ -226,7 +227,7 @@ void RSkComponentTextInput::processEventKey (rnsKey eventKeyType,bool* stopPropa
   auto textInputEventEmitter = std::static_pointer_cast<TextInputEventEmitter const>(component.eventEmitter);
   auto const &textInputProps = *std::static_pointer_cast<TextInputProps const>(component.props);
   keyPressMetrics.text = RNSKeyMap[eventKeyType];
-    if ((eventKeyType >= RNS_KEY_1 && eventKeyType <= RNS_KEY_z)) {
+    if ((eventKeyType >= RNS_KEY_1 && eventKeyType <= RNS_KEY_Less)) {
```
Comment (shefayaparwin) [inline ReactSkia/components/RSkComponentTextInput.cpp:232]: Can we have  Key_Type_Start & Key_TYPE_End enum to confine to pack the keys in groups .
Ex:-RNS_RCU_KEY_START,RCU_KEY_END,RNS_NAV_KEY_STRAT,RNS_NAV_KEY_END.
I feel, Which may give flexibility to modify the key support in future .

File: ReactSkia/sdk/RNSKeyCodeMapping.h [hunk 3]
```diff
@@ -33,9 +33,9 @@ typedef enum rnsKey{
     RNS_KEY_Guia,
     //Navigation Keys
     
+    RNS_KEY_Select,
```
Comment (shefayaparwin) [inline ReactSkia/sdk/RNSKeyCodeMapping.h:36]: Can we have Key_Type_Start & Key_TYPE_End enum to confine to pack the keys in groups .
Ex:-RNS_RCU_KEY_Start,RCU_KEY_End,RNS_NAV_KEY_STRAT,RNS_NAV_KEY_END,RNS_KB_KEY_START,RNS_KB_KEY_END
I feel, Which may give flexibility to modify the key support in future .

File: ReactSkia/sdk/RNSKeyCodeMapping.h [hunk 4]
```diff
@@ -148,18 +134,33 @@ typedef enum rnsKey{
...
-    RNS_KEY_Home,
-    RNS_KEY_EndKey,
-    RNS_KEY_Insert,
-    /*Arthematic */
+        /*Arthematic */
     RNS_KEY_Minus,
     RNS_KEY_Equal,
     RNS_KEY_Percent,
     RNS_KEY_Plus,
     RNS_KEY_Grater,
```
Comment (shefayaparwin) [inline ReactSkia/sdk/RNSKeyCodeMapping.h]: Look like typo error : RNS_KEY_Grater,->RNS_KEY_Greater

File: ReactSkia/sdk/RNSKeyCodeMapping.h [hunk 5]
```diff
@@ -148,18 +134,33 @@ typedef enum rnsKey{
...
     RNS_KEY_Less,
```
Comment (shefayaparwin) [inline ReactSkia/sdk/RNSKeyCodeMapping.h:144]: Minor comment : Can RNS_KEY_Less be changed to  RNS_KEY_Lesser to match with  RNS_KEY_Greater

Comment (chatrapathi06) [reply ReactSkia/sdk/RNSKeyCodeMapping.h:144]: RNS_KEY_Less is kept because the key name should be inline with libwpe and X11 key name. 
    Example X11:
    case XK_greater: return RNS_KEY_Greater;
    case XK_less: return RNS_KEY_Less;
   Example libWPE:
       case WPE_KEY_greater: return RNS_KEY_Greater;
    case WPE_KEY_less : return RNS_KEY_Less;

File: ReactSkia/components/RSkComponentTextInput.cpp (hunk 2 above)
Comment (chatrapathi06) [reply ReactSkia/components/RSkComponentTextInput.cpp:232]: discussed offline, unresolved

File: packages/react-native-skia/components/textinput/TextInput_Ref.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+const styles = StyleSheet.create({
+  input: {
+    height: 40,
+    margin: 12,
+    borderWidth: 1,
+    padding: 10,
+  },
+});
+
+export default SimpleViewApp;
```
Comment (pprabhu-ng) [inline packages/react-native-skia/components/textinput/TextInput_Ref.js]: Optional:
Is it possible to update the functionalities in this test app to the already existing test apps of text input component.. So that we can improve on additing functionalities in single app and it will be easy for regression tests as well.

**Comment by chatrapathi06** [comment]
Sure I will be add those in file

File: ReactSkia/core_modules/RSkSpatialNavigator.cpp [hunk 7]
```diff
@@ -394,17 +394,17 @@ bool RSkSpatialNavigator::advanceFocusInDirection(Container *container, rnsKey k
...
-
-  // Focus the candidate and update the spatial navigator states
-  if(currentFocus_) // First Blur the existing focus component
-    sendNotificationWithEventType("blur", currentFocus_->getComponentData().tag);
-  sendNotificationWithEventType("focus", focusCandidate->getComponentData().tag);
-  RNS_LOG_DEBUG("Blur : [" << ((currentFocus_) ? currentFocus_->getComponentData().tag : -1) << "], Focus :[" << focusCandidate->getComponentData().tag << "]");
-  currentFocus_ = focusCandidate;
-  currentContainer_ = currentFocus_->isContainer() ? currentFocus_ : currentFocus_->nearestAncestorContainer();
-
+  updateFocusCandidate(focusCandidate);
```
Comment (munezbn) [inline ReactSkia/core_modules/RSkSpatialNavigator.cpp:398]: Can you please the comment 
  // Focus the candidate and update the spatial navigator states

**Review by pprabhu-ng — APPROVED** [review]

**Review by munezbn — APPROVED** [review]

**Review by shefayaparwin — APPROVED** [review]
